### PR TITLE
feat: Run mdbook test in docs-book gha workflow

### PR DESCRIPTION
**Pull request by viccuad**
Signed-off-by: Víctor Cuadrado Juan <vcuadradojuan@suse.de>

### CODE REVIEW

File: .github/workflows/docs-book.yml [hunk 1]
```diff
@@ -15,10 +15,12 @@ jobs:
         uses: peaceiris/actions-mdbook@v1
         with:
           mdbook-version: '0.4.3'
+
       - run: mdbook build
+      - run: mdbook test
```
Comment (Itxaka) [inline .github/workflows/docs-book.yml:20]: does this actually works as expected? Running it locally it fails with a spectacular number of failures :o

Comment (viccuad) [reply .github/workflows/docs-book.yml:20]: you got me :>. I'm trying to run it locally though, and failing spectacularly..

Comment (viccuad) [reply .github/workflows/docs-book.yml:20]: also, for this to work we need to have [mdbook-linkcheck](https://github.com/Michael-F-Bryan/mdbook-linkcheck) and say so in the book.toml.

**Comment by viccuad** [comment]
Tried with the approach of downloading mdbook, mdbook-linkcheck binaries (and caching them).
See results here: https://github.com/viccuad/hypper/runs/2342661504

**Comment by viccuad** [comment]
tested the relative links with `mdbook serve`.

**Comment by viccuad** [comment]
Rebased to actually use gha cache. Example: https://github.com/viccuad/hypper/runs/2342910590?check_suite_focus=true#step:3:11

**Review by Itxaka — APPROVED** [review]

**Pull request merged**